modules/auxiliary/scanner/rdp/cve_2019_0708.rb

```diff
@@ -0,0 +1,545 @@
+    return [data].pack("H*")
+  end
+
+  def rdp_parse_serverdata(pkt)
+    ptr = 0
+    rdp_pkt = pkt[0x49..pkt.length]
+    
+    while ptr < rdp_pkt.length
+      header_type = rdp_pkt[ptr..ptr+1]
+      header_length = rdp_pkt[ptr+2..ptr+3].unpack("S<")[0]
+
+      vprint_status("header: #{bin_to_hex(header_type)} len #{header_length}")
+
+      if header_type == "\x02\x0c"
+        vprint_status("security header")
+
+        server_random = rdp_pkt[ptr+20..ptr+51]
+        public_exponent = rdp_pkt[ptr+84..ptr+87]
+        
+        modulus = rdp_pkt[ptr+88..ptr+151]
+        vprint_status("modulus_old #{bin_to_hex(modulus)}")
+
+        vprint_status("RSA magic: #{rdp_pkt[ptr+68..ptr+71]}")
+        bitlen = rdp_pkt[ptr+72..ptr+75].unpack("L<")[0] - 8
+        vprint_status("RSA bitlen: #{bitlen}")
+        modulus = rdp_pkt[ptr+88..ptr+87+bitlen]
+        vprint_status("modulus_new #{bin_to_hex(modulus)}")
+
+        
+      end
+
+
+      ptr += header_length
+    end
+
+
+
+    #server_random = hex_to_bin("3280b53a3cf38b7e8b4d9f3539524ba35455aea9ee974f910703534e6ddcb35c")
+
+    vprint_status("SERVER_MODULUS: #{bin_to_hex(modulus)}")
+    vprint_status("SERVER_EXPONENT: #{bin_to_hex(public_exponent)}")
+    vprint_status("SERVER_RANDOM: #{bin_to_hex(server_random)}")
+
+    rsmod = bytes_to_bignum(modulus)
+    rsexp = bytes_to_bignum(public_exponent)
+    rsran = bytes_to_bignum(server_random)
+
+    #vprint_status("MODULUS  = #{bin_to_hex(modulus)} - #{rsmod.to_s}")
+    #vprint_status("EXPONENT = #{bin_to_hex(public_exponent)} - #{rsexp.to_s}")
+    #vprint_status("SVRANDOM = #{bin_to_hex(server_random)} - #{rsran.to_s}")
+
+    return rsmod, rsexp, rsran, server_random, bitlen
+  end
+
+# used to abruptly abort scanner for a given host
+  class RdpCommunicationError < StandardError
+  end
+
+  def rdp_send(data)
+    sock.put(data)
+    #sock.flush
+    #sleep(0.1)
+    #sleep(0.5)
+  end
+
+  def rdp_recv()
+    res1 = sock.get_once(4, 5)
+    raise RdpCommunicationError unless res1 # nil due to a timeout
+    res2 = sock.get_once(res1[2..4].unpack("S>")[0], 5)
+    raise RdpCommunicationError unless res2 # nil due to a timeout
+    res1 + res2
+  end
+
+  def rdp_send_recv(data)
+    rdp_send(data)
+    return rdp_recv()
+  end
+
+
+  def rdp_encrypted_pkt(data, rc4enckey, hmackey, flags = "\x08\x00", flagsHi = "\x00\x00", channelId="\x03\xeb")
+    userData_len = data.length + 12
+    udl_with_flag = 0x8000 | userData_len
+
+    pkt = ""
+    pkt << "\x02\xf0\x80" # X.224
```

Comment:
remove line 223 and change to below
```suggestion
    pkt = "\x02\xf0\x80" # X.224
```

Reply:
fixed 3 instances of this